File: src/powerClient.ts
import { Gpio } from 'onoff';

export default class PowerClient {
    gpio: Gpio;

    constructor(pin: number) {
        this.gpio = new Gpio(pin, 'out');
    }

    turnOn() {
        this.gpio.writeSync(1);
    }

    turnOff() {
        this.gpio.writeSync(0);
    }

    read() {
        return this.gpio.readSync();
    }

    isOn() {
        return this.read() === 1;
    }
}

File: py/powerClient.py
from gpiozero import OutputDevice

class PowerClient:
    def __init__(self, pin):
        self.gpio = OutputDevice(pin)

    def turnOn(self):
        self.gpio.on()

    def turnOff(self):
        self.gpio.off()

    def read(self):
        return self.gpio.value

    def isOn(self):
        return self.read() == 1
